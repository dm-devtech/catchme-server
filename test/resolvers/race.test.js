const gql = require("graphql-tag");
const { ApolloServer } = require("apollo-server");
const resolvers = require("../../src/resolvers/race.js");
const typeDefs = require("../../src/schema");
const models = require("../../models");
const { createTestClient } = require("apollo-server-testing");
const db = require("../../models/index");

const CREATE_RACE = gql`
  mutation createRace($distance: Int!) {
    createRace(distance: $distance) {
      id
      distance
    }
  }
`;
const QUERY_RACE = gql`
  query race($id: Int!) {
    race(id: $id) {
      id
      distance
    }
  }
`;
describe("Race resolvers", () => {
  let server;
  beforeAll(async () => {
    await db.sequelize.sync({ force: true });
  });

  beforeEach(async () => {
    server = new ApolloServer({
      typeDefs,
      resolvers,
      context: { models },
    });
  });

  afterAll(async () => {
    await db.sequelize.close();
  });
  afterEach(async () => {
    await server.stop();
  });

  it("fetches single race", async () => {
    const { query, mutate } = createTestClient(server);
    const res = await mutate({
      mutation: CREATE_RACE,
      variables: { distance: 1000 },
    });
    expect(res).toMatchSnapshot();
  });

  it("fetches single race", async () => {
    const { query, mutate } = createTestClient(server);
    const res = await query({
      query: QUERY_RACE,
      variables: { id: 1 },
    });
    expect(res).toMatchSnapshot();
    expect(res.data).toEqual({ race: { distance: 1000, id: 1 } });
    expect(res.error).toEqual(undefined);
  });
});
