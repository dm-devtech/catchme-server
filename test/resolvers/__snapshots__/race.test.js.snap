// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Race resolvers fetches single race 1`] = `
Object {
  "data": Object {
    "createRace": Object {
      "distance": 1000,
      "id": 1,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Race resolvers fetches single race 2`] = `
Object {
  "data": Object {
    "race": Object {
      "distance": 1000,
      "id": 1,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
